src/routes/feeds.rs

```diff
@@ -75,11 +76,34 @@ async fn get_all_feed(
   )
 }
 
-fn get_feed_all_data(conn: &PgConnection, sort_type: &SortType) -> Result<String, LemmyError> {
+async fn get_local_feed(
+  info: web::Query<Params>,
+  context: web::Data<LemmyContext>,
+) -> Result<HttpResponse, Error> {
+  let sort_type = get_sort_type(info).map_err(ErrorBadRequest)?;
+
+  let rss = blocking(context.pool(), move |conn| {
+    get_feed_data(conn, &ListingType::Local, &sort_type)
+  })
+  .await?
+  .map_err(ErrorBadRequest)?;
+
+  Ok(
+    HttpResponse::Ok()
+      .content_type("application/rss+xml")
+      .body(rss),
+  )
+}
```

Comment:
I managed to rewrite this so that get_local_feed and get_all_feed dont duplicate all the same logic. But its a bit stupid because it still needs two different functions, just to set the ListingType.

https://github.com/LemmyNet/lemmy/commit/54d3a4d4cea7248b3fa9119beb17f9e8615f8ceb

Reply:
Yeah its kind of a mess because of the different routes. Oh well. I updated.